lib/matplotlib/axes/_base.py

```diff
@@ -2970,6 +2970,9 @@ def set_axis_off(self):
         """Turn off the axis."""
         self.axison = False
         self.stale = True
+        for k in self.spines.keys():
+            self.spines[k].set_visible(False)
+
```

Comment:
to be pedantic, this should be done prior to setting the stale flag to True.

Reply:
Ooops. I meant to remove that anyhow..